### DISCUSSION TITLE
DMX to OSC

### DISCUSSION THREAD

**Discussion by ggtimtom** [sstaub/gma3#8] [Ideas]
Hello Stefan,

thank you for this great project! What do you think about building a DMX-to-OSC Adapter ?
Using a [LTC485](https://www.analog.com/media/en/technical-documentation/data-sheets/485fm.pdf) chip to bring DMX to an serialport of your microcontroller. 
Then every cheap DMX-Controller can be used as Hardware to control Grandma3.
https://www.thomann.de/de/dmx_lichtsteuerpulte.html?oa=pra&gk=LISTDM&cme=false&filter=true

Regards,
Günter

**Comment by sstaub** [comment]
Personally I think the idea is not good. Maybe you buy a cheap lighting console and modify it. 
But you can try with Wiznet W5500-EVB-Pico, there is also a good dmx library for Pico available.

**Reply by ggtimtom** [reply]
OK thanks. But i think its easier for me to write the DMX-Code for the STM32 than modify a DMX-Console.

**Comment by sstaub** [comment]
Ok, you should avoid sending data if not necessary and use only UDP.

**Comment by ggtimtom** [comment]
I think its the same as when i use real faders. Check if the value has changed, then write to OSC.
